doc/source/development.rst

```diff
@@ -28,162 +32,166 @@ Fork the upstream KIWI repository
-the next step is to create the virtual development environment.
-The following procedure describes how to create a Python3 virtual
-development environment:
+On Fedora the following commands should install the required packages:
 
-1. Create the virtual environment:
+.. code-block:: shell-session
 
-   .. code:: bash
+   $ dnf install python3 python3-devel 'python3dist(pip)' \
+      'python3dist(tox)' make gcc which xz xorriso libxml2-devel \
+      libxslt-devel enchant genisoimage ShellCheck
+   $ # LaTeX packages
+   $ dnf install latexmk texlive-cmap texlive-metafont texlive-ec \
+      texlive-babel-english texlive-fncychap texlive-fancyhdr \
+      texlive-titlesec texlive-tabulary texlive-framed texlive-wrapfig \
+      texlive-parskip texlive-upquote texlive-capt-of texlive-needspace \
+      texlive-makeindex texlive-times texlive-helvetic texlive-courier \
+      texlive-gsftopk texlive-updmap-map texlive-dvips
 
-    $ python3 -m venv .env3
 
-2. Activate the virtual environment:
+Create a python virtual development environment
+-----------------------------------------------
 
-   .. code:: bash
+We recommend to setup a Python virtualenv for development. For Python 3
+:command:`pyvenv` is used, as it is bundled with Python itself (see
+https://docs.python.org/3.3/whatsnew/3.3.html#pep-405-virtual-environments
+for details). For Python 2.7 use :command:`virtualenv`, which is provided
+via pip or as an extra package by your favorite Linux distribution.
 
-    $ source .env3/bin/activate
+The following commands will setup an environment for Python 3:
 
-3. Install KIWI requirements inside the virtual environment:
+.. code-block:: shell-session
 
-   .. code:: bash
+   $ python3 -m venv .env3
+   $ source .env3/bin/activate
+   $ pip install -r .virtualenv.dev-requirements.txt
+   $ python3 setup.py develop
 
-    $ pip install -r .virtualenv.dev-requirements.txt
+The :command:`develop` target of the :command:`setup.py` script
+automatically creates the application entry point called
+:command:`kiwi-ng-3`, which allows you to call the your modified version of
+KIWI from inside the virtual environment:
 
-4. Install KIWI in Development Mode:
+.. code-block:: shell-session
 
-   .. code:: bash
+    $ kiwi-ng-3 --help
 
-     $ ./setup.py develop
+.. warning::
 
-Once the development environment is activated and initialized with the
-project required Python modules, you are ready to work.
+   The virtualenv's `$PATH` will not be taken into account when calling
+   KIWI via :command:`sudo`! Use the absolute path to the KIWI executable
+   to run an actual build using your local changes:
 
-The :command:`develop` target of the :command:`setup.py` script
-automatically creates the application entry point called :command:`kiwi-ng-3`,
-which allows to simply call the application from the code in the
-virtual environment:
+   .. code-block:: shell-session
 
-.. code:: bash
+      $ /path/to/my/kiwi_local_copy/.env3/bin/kiwi-ng-3 system build ...
```

Comment:
Isn't a `sudo` command missing here?

Or -- to simplify the line a bit -- you could use `$PWD` (assuming you are in the right directory):

```
$ sudo $PWD/.env3/bin/kiwi-ng-3 system build ...
```